tests/std/tests/P2165R4_tuple_like_pair/test.cpp

```diff
@@ -0,0 +1,189 @@
+    static_assert(constructible_from<pair<const int&&, const int&&>, const PairLike<int>>);
+
+    PairLike<TmpInt> a = {1, 2};
+
+    pair<TmpInt, TmpInt> p1(a);
+    assert(p1.first == 1 && p1.second == 2 && get<0>(a) == 1 && get<1>(a) == 2);
+
+    pair<TmpInt&, TmpInt&> p2(a);
+    assert(&p2.first == &get<0>(a) && &p2.second == &get<1>(a));
+
+    pair<const TmpInt&, const TmpInt&> p3(a);
+    assert(&p3.first == &get<0>(a) && &p3.second == &get<1>(a));
+
+    pair<TmpInt, TmpInt> p4(as_const(a));
+    assert(p4.first == 1 && p4.second == 2 && get<0>(a) == 1 && get<1>(a) == 2);
+
+    pair<const TmpInt&, const TmpInt&> p5(as_const(a));
+    assert(&p5.first == &get<0>(a) && &p5.second == &get<1>(a));
+
+    pair<TmpInt, TmpInt> p6(std::move(a));
+    assert(p6.first == 1 && p6.second == 2 && get<0>(a) == -1 && get<1>(a) == -1);
+    tie(get<0>(a), get<1>(a)) = std::move(p6);
+
+    pair<const TmpInt&, const TmpInt&> p7(std::move(a));
+    assert(&p7.first == &get<0>(a) && &p7.second == &get<1>(a));
+
+    pair<TmpInt&&, TmpInt&&> p8(std::move(a));
+    assert(&p8.first == &get<0>(a) && &p8.second == &get<1>(a));
+
+    pair<const TmpInt&&, const TmpInt&&> p9(std::move(a));
+    assert(&p9.first == &get<0>(a) && &p9.second == &get<1>(a));
+
+    pair<TmpInt, TmpInt> p10(std::move(as_const(a)));
+    assert(p10.first == 1 && p10.second == 2 && get<0>(a) == 1 && get<1>(a) == 2);
+
+    pair<const TmpInt&, const TmpInt&> p11(std::move(as_const(a)));
+    assert(p11.first == 1 && p11.second == 2 && get<0>(a) == 1 && get<1>(a) == 2);
+
+    pair<const TmpInt&&, const TmpInt&&> p12(std::move(as_const(a)));
+    assert(p12.first == 1 && p12.second == 2 && get<0>(a) == 1 && get<1>(a) == 2);
+
+    return true;
+}
+
+template <template <class> class PairLike>
+constexpr bool test_pair_like_assignment() {
+    static_assert(is_assignable_v<pair<int, int>&, PairLike<int>&>);
+    static_assert(is_assignable_v<pair<int, int>&, const PairLike<int>&>);
+    static_assert(is_assignable_v<pair<int, int>&, PairLike<int>>);
+    static_assert(is_assignable_v<pair<int, int>&, const PairLike<int>>);
+
+    PairLike<TmpInt> a = {1, 2};
+
+    pair<TmpInt, TmpInt> p1;
+    p1 = a;
+    assert(p1.first == 1 && p1.second == 2 && get<0>(a) == 1 && get<1>(a) == 2);
+
+    pair<TmpInt, TmpInt> p2;
+    p2 = as_const(a);
+    assert(p2.first == 1 && p2.second == 2 && get<0>(a) == 1 && get<1>(a) == 2);
+
+    pair<TmpInt, TmpInt> p3;
+    p3 = std::move(a);
+    assert(p3.first == 1 && p3.second == 2 && get<0>(a) == -1 && get<1>(a) == -1);
+    tie(get<0>(a), get<1>(a)) = std::move(p3);
+
+    pair<TmpInt, TmpInt> p4;
+    p4 = std::move(as_const(a));
+    assert(p4.first == 1 && p4.second == 2 && get<0>(a) == 1 && get<1>(a) == 2);
+
+    return true;
+}
+
+template <template <class> class PairLike>
+constexpr bool test_pair_like_const_assignment() {
+    using Ref  = std::vector<bool>::reference;

```

Comment:
Qualifying `std::vector` is unnecessary as this test is `using namespace std;`. Occurs below.